src/marqo/tensor_search/tensor_search.py

```diff
@@ -1906,9 +1943,15 @@ def get_query_vectors_from_jobs(
             merged_vector = np.mean(weighted_vectors, axis=0)
 
             if q.index.normalize_embeddings:
-                norm = np.linalg.norm(merged_vector, axis=-1, keepdims=True)
-                if norm > 0:
-                    merged_vector /= np.linalg.norm(merged_vector, axis=-1, keepdims=True)
+                try:
+                    merged_vector = normalize_vector(merged_vector)
+                except core_exceptions.ZeroMagnitudeVectorError as e:
+                    tb = traceback.extract_tb(e.__traceback__)[-1]
+                    raise core_exceptions.ZeroMagnitudeVectorError(
+                        f"Zero magnitude vector detected at file name: {tb.filename}, line number: {tb.lineno}, inside function name: {tb.name}. "
```

Comment:
`Zero magnitude vector detected at file name:` isn't a very intuitive error message I think. It could be something like `Magnitude of combined query and context vectors cannot be zero...`, then The filename and other details can be afterwards.

Reply:
Agreed. The message must be user friendly as it goes to the api. api exception handler will log the stack trace so don't worry about line number etc